tests/libvmi/factory.go

```diff
@@ -34,14 +34,30 @@ const (
 // NewFedora instantiates a new Fedora based VMI configuration,
 // building its extra properties based on the specified With* options.
 func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
-	return newFedora(cd.ContainerDiskFedora, opts...)
+	return newFedoraWithUserDataSetPassword(cd.ContainerDiskFedora, opts...)
```

Comment:
This new `newFedoraWithUserDataSetPassword` is not needed, you should just have:
```
newFedora(cd.ContainerDiskFedora, WithCloudInitNoCloudUserData(userDataSetPassword, false), opts...)
```
Where `userDataSetPassword is a constant`.

Reply:
```
too many arguments in call to newFedora
	have (tests.ContainerDisk, Option, []Option...)
	want (tests.ContainerDisk, ...Option)go
```
	
I can append it in place but then i will need to do it twice (this is why i added a function),
or if there is a more elegant way ? but specifically what you suggested gives an error.

Reply:
Ahh, right. So just append it in a one-liner before.
What exactly you need to do twice?

Reply:
do this append twice, because there are two functions that need it
NewFedora
NewTestToolingFedora